requirements.txt

```diff
@@ -41,3 +41,5 @@ transformers>=4.40.0 ; python_version == '3.10'
 transformers>=4.41.2 ; python_version >= '3.11'
 trl>=0.9.4
 wandb>=0.16.4
+nvidia-ml-py==12.555.43
+nvidia_smi==0.1.3
```

Comment:
Why these libs?
nvidia_smi is not maintained very well, doesn't seem to have a link to source.
The nvidia-ml-py is better, but still, the same information could be obtained with torch, which is already in deps.

And as a bonus, it would allow for integration with other providers of high end ML HW. Like TPUs from Google.

```
for i in range(torch.cuda.device_count()):
  torch.cuda.get_device_properties(i)
_CudaDeviceProperties(name='Tesla T4', major=7, minor=5, total_memory=14915MB, multi_processor_count=40)
```

https://pytorch.org/docs/stable/cuda.html#module-torch.cuda

Reply:
thanks! switched to this